### PR TITLE
[CPU] Prohibit to load model with dynamic output shapes for legacy API

### PR DESCRIPTION
**Pull request by mandrono**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/plugins/intel_cpu/src/infer_request.cpp [hunk 1]
```diff
@@ -390,6 +390,11 @@ void ov::intel_cpu::MKLDNNLegacyInferRequest::SetBlob(const std::string& name, c
     findInputAndOutputBlobByName(name, foundInput, foundOutput);
 
     if (foundInput) {
+        const auto inputNode = graph->getInputNodeByName(name);
+        if (inputNode->getOutputShapeAtPort(0).isDynamic()) {
+            IE_THROW() << "Can't set input blob for dynamic model via legacy API!";
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/infer_request.cpp:395]:
```suggestion
            IE_THROW(Unexpected) << "Can't set input blob for a model with dynamic shapes via legacy API!";
```

File: src/plugins/intel_cpu/src/infer_request.cpp [hunk 2]
```diff
@@ -442,6 +447,11 @@ void ov::intel_cpu::MKLDNNLegacyInferRequest::SetBlob(const std::string& name, c
         }
     }
     if (foundOutput) {
+        const auto outputNode = graph->getOutputNodeByName(name);
+        if (outputNode->getInputShapeAtPort(0).isDynamic()) {
+            IE_THROW() << "Can't set output blob for dynamic model via legacy API!";
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/infer_request.cpp:452]:
```suggestion
            IE_THROW(Unexpected) << "Can't set output blob for a model with dynamic shapes via legacy API!";
```

File: src/plugins/intel_cpu/src/infer_request.cpp [hunk 3]
```diff
@@ -490,6 +500,10 @@ InferenceEngine::Blob::Ptr ov::intel_cpu::MKLDNNLegacyInferRequest::GetBlob(cons
     const auto &inMap = graph->inputNodesMap;
     auto input = inMap.find(name);
     if (input != inMap.end()) {
+        if (input->second->getOutputShapeAtPort(0).isDynamic()) {
+            IE_THROW() << "Can't get input blob for dynamic model via legacy API!";
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/infer_request.cpp:504]:
```suggestion
            IE_THROW(Unexpected) << "Can't get input blob for a model with dynamic shapes via legacy API!";
```

File: src/plugins/intel_cpu/src/infer_request.cpp [hunk 4]
```diff
@@ -542,8 +556,13 @@ InferenceEngine::Blob::Ptr ov::intel_cpu::MKLDNNLegacyInferRequest::GetBlob(cons
     }
 
     if (graph->hasOutputWithName(name)) {
+        const auto outputNode = graph->getOutputNodeByName(name);
+        if (outputNode->getInputShapeAtPort(0).isDynamic()) {
+            IE_THROW() << "Can't get output blob for dynamic model via legacy API!";
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/infer_request.cpp:561]:
```suggestion
            IE_THROW(Unexpected) << "Can't get output blob for a model with dynamic shapes via legacy API!";
```

File: src/plugins/intel_cpu/src/plugin.cpp [hunk 5]
```diff
@@ -676,6 +676,16 @@ Engine::LoadExeNetworkImpl(const InferenceEngine::CNNNetwork &network, const std
     auto nGraphFunc = clonedNetwork.getFunction();
     TransformationUpToCPUSpecificOpSet(nGraphFunc, enableLPT, enableSnippets, isLegacyAPI());
 
+    // need to check that all outputs have static shapes
+    // checking that all inputs have static shapes perform in common part
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/plugin.cpp:680]:
```suggestion
    // checking that all inputs have static shapes is performed in the common part
```

File: src/plugins/intel_cpu/src/plugin.cpp [hunk 6]
```diff
@@ -676,6 +676,16 @@ Engine::LoadExeNetworkImpl(const InferenceEngine::CNNNetwork &network, const std
...
+    if (isLegacyAPI()) {
+        for (const auto& res : nGraphFunc->get_results()) {
+            if (res->get_input_partial_shape(0).is_dynamic()) {
+                IE_THROW() << "CPU plug-in can't load model with dynamic output shapes via legacy API.";
```
Comment (maxnick) [inline src/plugins/intel_cpu/src/plugin.cpp:684]:
```suggestion
                IE_THROW() << "CPU plug-in can't load a model with dynamic output shapes via legacy API.";
```

**Review by maxnick — APPROVED** [review]

**Review by tgubanova-lohika — CHANGES_REQUESTED** [review]
Stop russian aggression in Ukraine!

**Pull request merged**